manifests/init.pp

```diff
@@ -44,6 +48,8 @@
   $odl_rest_port       = $::opendaylight::params::odl_rest_port,
   $odl_bind_ip         = $::opendaylight::params::odl_bind_ip,
   $rpm_repo            = $::opendaylight::params::rpm_repo,
+  $baseurl             = "http://cbs.centos.org/repos/nfv7-${rpm_repo}/\$basearch/os/",
```

Comment:
I don't think this will work.  rpm_repo is always reading the one inherited from params because the rpm_repo you are overriding in this class hasn't been parsed yet.  So it is never using the value you pass to rpm_repo.  One option is to just make baseurl a separate param that is inherited, like this:
  $baseurl             = $::opendaylight::params::baseurl,

params.pp:
  $baseurl = "http://cbs.centos.org/repos/nfv7-${rpm_repo}/\$basearch/os/"

Then the user is responsible to override baseurl as well as rpm_repo if they want to change how the repo is created.